<a href="https://colab.research.google.com/github/bbishal12/Quiz0/blob/main/Quiz5_SQL_Sub_Queries.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Data Visualization
##Bishal Budhathoki
##Quiz 5 : SQL Sub Queries

Use the Cloud Resource Manager to Create a Cloud Platform project if you do not already have one. Enable billing for the project. Enable BigQuery APIs for the project.

In [2]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Now that we are authenticated. We are going to look at the bikers trip data from Austin. We have a project called data-viz-bishal in my BigQuery interface.

1. What was the second most popular starting station?

2. How many trips lasted over an hour and were a round trip (started and stopped at the same station)?


In [6]:
%%bigquery --project data-viz-bishal
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
LIMIT 10

trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19
5  9900286143  ...               17
6  9900286171  ...                6
7  9900286214  ...                0
8  9900286540  ...               12
9    13575843  ...               47

[10 rows x 9 columns]

The above dataset is the data from bikershare trips of start time and end time with start station and end station.

In [7]:
%%bigquery --project data-viz-bishal
SELECT MAX(start_station_name) as second_most_popular
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name != (SELECT MAX(start_station_name) as most_popular_start_station FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`)

second_most_popular
0  Zilker Park at Barton Springs & William Barton...

The most second popular starting station is Zilker Park at Barton Springs and William Barton.

In [8]:
%%bigquery --project data-viz-bishal
SELECT  COUNTIF(duration_minutes>=60) as over_an_hour_duration ,COUNT( start_station_name) as number_of_trips
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name

over_an_hour_duration  number_of_trips
0                  56369           228339

228339 trips lasted over an hour and were a round trip (started and stopped at the same station).

In [9]:
%%bigquery --project data-viz-bishal
SELECT  start_station_name as station_name,  COUNTIF(duration_minutes>=60) as over_an_hour_duration, COUNT( start_station_name) as number_of_trips
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name
GROUP BY start_station_name

station_name  ...  number_of_trips
0                    Toomey Rd @ South Lamar  ...              633
1            State Capitol @ 14th & Colorado  ...              910
2                           Waller & 6th St.  ...              583
3                                 Pease Park  ...              167
4    Capital Metro HQ - East 5th at Broadway  ...              635
..                                       ...  ...              ...
179    Republic Square @ Guadalupe & 4th St.  ...             1090
180                               3rd & West  ...             2002
181                                 3rd/West  ...              816
182                              26th/Nueces  ...              607
183                            Nueces & 26th  ...             1425

[184 rows x 3 columns]

The above dataset is the calculation for the ways to get trips lasted over an hour and for the round trip.